### PR TITLE
Remove generated translation binaries from repo

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/jmoiron/humanize/issues/198

Changes proposed in this pull request:

* Remove the `.mo` files from the repo and .gitignore them
* Add a script to generate them (requires gettext/msgfmt)
* Install gettext/msgfmt and and generate files on CI for Linux and macOS
* It's a bit harder to install on CI for Windows, skip installation there
* Skip tests requiring `.mo` files if not found, the tester may not have gettext/msgfmt (like Windows CI)
* Add step to release checklist to generate `.mo` files

### CODE REVIEW

**Comment by hugovk** [comment]
Python 3.10-dev failures are due to GHA using some sort of 3.10 beta, but 3.10 isn't actually in beta yet, and pytest isn't quote yet ready for 3.10 beta:

* https://github.com/actions/setup-python/issues/207
* https://github.com/pytest-dev/pytest/pull/8540#issuecomment-822330273

Will ignore for now, should be resolved soon enough.

File: scripts/generate-translation-binaries.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+set -e
+
+for d in src/humanize/locale/*/; do
+    locale="$(basename $d)"
+    echo "$locale"
+    # compile to binary .mo
+    msgfmt --check -o src/humanize/locale/$locale/LC_MESSAGES/humanize{.mo,.po}
+done
```
Comment (jordimas) [inline scripts/generate-translation-binaries.sh:8]: My only comment is that if msgfmt --check fails, for example if in the translation of the string "%d months" you forget the "%d", then this should stop and get fixed. You may need to check the exit code and act accordinly.

**Comment by jordimas** [comment]
Looks good to me. Thanks

File: scripts/generate-translation-binaries.sh (hunk 1 above)
Comment (hugovk) [reply scripts/generate-translation-binaries.sh:8]: Yes, good point. I checked it in another branch. A change like this:

```diff
diff --git a/src/humanize/locale/ca_ES/LC_MESSAGES/humanize.po b/src/humanize/locale/ca_ES/LC_MESSAGES/humanize.po
index 5e6a4f2..abdc464 100644
--- a/src/humanize/locale/ca_ES/LC_MESSAGES/humanize.po
+++ b/src/humanize/locale/ca_ES/LC_MESSAGES/humanize.po
@@ -221,8 +221,8 @@ msgstr "un mes"
 #, python-format
 msgid "%d month"
 msgid_plural "%d months"
-msgstr[0] "%d mes"
-msgstr[1] "%d mesos"
+msgstr[0] "mes"
+msgstr[1] "mesos"

 #: src/humanize/time.py:174
 msgid "a year"
```

https://github.com/hugovk/humanize/commit/cfa2788a72ef0efa12ff7390fac7b08dea12bf46

Fails on the CI like this:

```
Run scripts/generate-translation-binaries.sh
ca_ES
src/humanize/locale/ca_ES/LC_MESSAGES/humanize.po:224: number of format specifications in 'msgid_plural' and 'msgstr[0]' does not match
msgfmt: found 2 fatal errors
Error: Process completed with exit code 1.
```

https://github.com/hugovk/humanize/runs/2431670085?check_suite_focus=true

Comment (jordimas) [reply scripts/generate-translation-binaries.sh:8]: This is great. Thanks!

**Pull request merged**